src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/Public.Data/IPreviewTokenService.cs

```diff
@@ -0,0 +1,18 @@
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels.Public.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Public.Data;
+
+public interface IPreviewTokenService
+{
+    Task<Either<ActionResult, PreviewTokenViewModel>> CreatePreviewToken(
+        Guid dataSetVersionId, string label, CancellationToken cancellationToken = default);
```

Comment:
Is there a reason that you know of why we use the convention `CancellationToken cancellationToken = default`?  Would we ever want to allow our code to *not* pass a cancellationToken?

Reply:
Formatting might look neater with parameters on new lines?

Reply:
> Is there a reason that you know of why we use the convention CancellationToken cancellationToken = default? Would we ever want to allow our code to not pass a cancellationToken?

I've made it mandatory on all private methods, expecting it to exist and be passed down into them, and made it optional on public methods, not knowing how they intend to be used.

I'm not sure of the answer, but I can't think of anything other than that using it isn't mandatory if the caller doesn't support it or doesn't want to use it.

I can think of a situation in unit testing where a `CancellationToken` is not available and repeating `CancellationToken.None` would get pretty repetitive.

Reply:
Good point, ta!